libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -4819,20 +4814,42 @@ MAV_RESULT GCS_MAVLINK::handle_command_long_packet(const mavlink_command_long_t
         result = handle_flight_termination(packet);
         break;
 
-    case MAV_CMD_COMPONENT_ARM_DISARM:
-        result = handle_command_component_arm_disarm(packet);
-        break;
-
-    case MAV_CMD_FIXED_MAG_CAL_YAW:
-        result = handle_fixed_mag_cal_yaw(packet);
-        break;
-
 #if AP_LANDINGGEAR_ENABLED
         case MAV_CMD_AIRFRAME_CONFIGURATION:
             result = handle_command_airframe_configuration(packet);
             break;
 #endif
 
+    case MAV_CMD_DO_SET_ROI_LOCATION:
+    case MAV_CMD_DO_SET_ROI:
+    case MAV_CMD_FIXED_MAG_CAL_YAW:
+    case MAV_CMD_COMPONENT_ARM_DISARM: {
+        // try the same thing as a COMMAND_INT:
+        MAV_FRAME frame = MAV_FRAME_GLOBAL_RELATIVE_ALT;
+        if (command_long_stores_location((MAV_CMD)packet.command)) {
+            switch (packet.command) {
+            case MAV_CMD_DO_SET_HOME:
+                // we should already have handled this command
+                frame = MAV_FRAME_GLOBAL;
+                break;
+            case MAV_CMD_DO_SET_ROI:
+            case MAV_CMD_DO_SET_ROI_LOCATION:
+                frame = MAV_FRAME_GLOBAL_RELATIVE_ALT;
+                break;
+            case MAV_CMD_NAV_TAKEOFF:
+                // can't determine a frame here as it may be above-home or above-terrain depending on vehicle.
+                return MAV_RESULT_UNSUPPORTED;
+            case MAV_CMD_DO_REPOSITION:
+                // we have never accepted this via COMMAND_LONG - don't start
+                return MAV_RESULT_UNSUPPORTED;
+            default:
+                return MAV_RESULT_UNSUPPORTED;
+            }
+        }
+        mavlink_command_int_t command_int;
+        convert_COMMAND_LONG_to_COMMAND_INT(packet, command_int, frame);
+        return handle_command_int_packet(command_int);
```

Comment:
having a table would allow for much simpler code in the default: case

Reply:
Changed to using a table (and to just call a method here rather than in-lining the code)